test/spec/openapi/option.js

```diff
@@ -344,4 +344,154 @@ test('cache - yaml', t => {
   })
 })
 
+test('transforms examples in example if single string example', t => {
+  t.plan(4)
+  const fastify = Fastify()
+
+  fastify.register(fastifySwagger, openapiOption)
+
+  const opts = {
+    schema: {
+      body: {
+        type: 'object',
+        required: ['hello'],
+        properties: {
+          hello: {
+            type: 'string',
+            examples: ['world']
+          }
+        }
+      }
+    }
+  }
+
+  fastify.get('/', opts, () => {})
+
+  fastify.ready(err => {
+    t.error(err)
+
+    const openapiObject = fastify.swagger()
+    const schema = openapiObject.paths['/'].get.requestBody.content['application/json'].schema
+
+    t.ok(schema)
+    t.notOk(schema.properties.hello.examples)
+    t.equal(schema.properties.hello.example, 'world')
+  })
+})
+
+test('transforms examples in example if single object example', t => {
+  t.plan(4)
+  const fastify = Fastify()
+
+  fastify.register(fastifySwagger, openapiOption)
+
+  const opts = {
+    schema: {
+      body: {
+        type: 'object',
+        required: ['hello'],
+        properties: {
+          hello: {
+            type: 'object',
+            properties: {
+              lorem: {
+                type: 'string'
+              }
+            },
+            examples: [{ lorem: 'ipsum' }]
+          }
+        }
+      }
+    }
+  }
+
+  fastify.get('/', opts, () => {})
+
+  fastify.ready(err => {
+    t.error(err)
+
+    const openapiObject = fastify.swagger()
+    const schema = openapiObject.paths['/'].get.requestBody.content['application/json'].schema
+
+    t.ok(schema)
+    t.notOk(schema.properties.hello.examples)
+    t.same(schema.properties.hello.example, { lorem: 'ipsum' })
+  })
+})
+
+test('uses examples if has multiple string examples', t => {
+  t.plan(4)
+  const fastify = Fastify()
+
+  fastify.register(fastifySwagger, openapiOption)
+
+  const opts = {
+    schema: {
+      body: {
+        type: 'object',
+        required: ['hello'],
+        properties: {
+          hello: {
+            type: 'string',
+            examples: ['hello', 'world']
```

Comment:
According to the document it should convert to something like
```js
{
  examples: {
    hello: {
      value: 'hello'
    },
    world: {
      value: 'world'
    }
  }
}
```

Reply:
Nice catch, totally missed it. To handle these cases I created a function `convertExamplesArrayToObject` that iterates through the `examples` array and convert it to an object using a `reduce`.

Created unit tests for `strings` and `number` examples and they had these outputs

Strings
```js
{
  hello: { value: 'hello' },
  world: { value: 'world' }
}
```

Numbers
```js
{
  1: { value: 2 },
  2: { value: 2 }
}
```

You think this solution solves this?